radicle-node/src/worker.rs

```diff
@@ -153,27 +160,19 @@ impl<G: Signer + Ecdh + 'static> Worker<G> {
         drain: Vec<u8>,
         mut session: WireSession<G>,
     ) -> (WireSession<G>, Result<Vec<RefUpdate>, FetchError>) {
-        let rid = fetch.rid;
         match &fetch.direction {
             FetchDirection::Initiator { namespaces } => {
                 log::debug!(target: "worker", "Worker processing outgoing fetch for {}", fetch.rid);
+                let staging = match fetch::StagingRad::new(&self.storage, fetch.rid) {
+                    Ok(staging) => staging,
+                    Err(err) => return (session, Err(err.into())),
+                };
```

Comment:
How would you feel about moving this into the 'fetch(..)'?  It would reduce the fetch() arguments to commonly types.  StagingRad and StagingRefs appear to be stages of the fetch action.